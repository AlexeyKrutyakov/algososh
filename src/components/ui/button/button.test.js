import TestRenderer from 'react-test-renderer';

import { Button } from './button';

describe('Рендеринг кнопки', () => {
  it('Кнопка с текстом рендерится без ошибок', () => {
    const tree = TestRenderer.create(
      <Button
        type="submit"
        text="Развернуть"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Кнопка без текста рендерится без ошибок', () => {
    const tree = TestRenderer.create(
      <Button
        type="submit"
        text="Развернуть"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Заблокированная кнопка рендерится без ошибок', () => {
    const tree = TestRenderer.create(
      <Button
        type="submit"
        text="Развернуть"
        disabled="true"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Кнопка с индикацией загрузки рендерится без ошибок', () => {
    const tree = TestRenderer.create(
      <Button
        type="submit"
        text="Развернуть"
        isLoader={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
