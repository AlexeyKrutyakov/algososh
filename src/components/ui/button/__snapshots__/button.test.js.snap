// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендеринг кнопки Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled="true"
  type="submit"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Рендеринг кнопки Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="submit"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Рендеринг кнопки Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="submit"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Рендеринг кнопки Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="submit"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;
